### ISSUE TITLE
Changes to Notifications seg faults

### ISSUE THREAD

**Issue by AndyM48** [martinrotter/rssguard#1391]
### Brief description of the issue

Any changes to Settings>Notfications seg faults on OK/Apply

### How to reproduce the bug?

Start rssguard
Select Settings>Notifications
Select any element eg Custom Notification, Select/Deselect any Balloon notification
Select OK or Apply

### What was the expected result?

Change should be registered and applied

### What actually happened?

rssguard segfaults, change is not applied

### Debug log

$ rssguard
time="     0.000" type="debug" -> core: Starting RSS Guard 4.6.6.
time="     0.000" type="debug" -> core: Current UTC date/time is 2024-04-29T09:43:19Z.
time="     0.296" type="debug" -> core: Initializing settings in '/home/andrew/.config/RSS Guard 4/config/config.ini' (non-portable way).
time="     0.296" type="debug" -> core: Starting to load active localization. Desired localization is 'en_GB'.
time="     0.298" type="debug" -> core: Application localization 'en_GB' loaded successfully, specifically sublocalization 'en_GB' was loaded.
time="     0.298" type="warning" -> core: Qt localization 'en_GB' WAS NOT loaded successfully.
time="     0.826" type="debug" -> database: File-based SQLite database connection 'DatabaseFactory' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     0.826" type="debug" -> database: File-based SQLite database has version '8'.
time="     0.827" type="debug" -> gui: Available icon theme paths: QList(/home/andrew/.icons, /home/andrew/.local/share/icons, /usr/share/icons, :/icons, :/graphics, /home/andrew/.config/RSS Guard 4/icons, /usr/bin/icons)
time="     0.829" type="debug" -> gui: Installed icon themes are: '', 'default', 'XCursor-Pro-Dark', 'custom', 'pixel-perfect-folder-icons', 'hicolor', 'elementary-xfce-darkest', 'elementary-xfce-darker', 'elementary-xfce', 'elementary-xfce-dark', 'Adwaita', 'oxygen', 'HighContrast', 'elementary', 'retrosmart-xcursor-white', 'retrosmart-xcursor-white-shadow', 'retrosmart-xcursor-white-color', 'retrosmart-xcursor-white-color-shadow', 'retrosmart-xcursor-black', 'retrosmart-xcursor-black-shadow', 'retrosmart-xcursor-black-color', 'retrosmart-xcursor-black-color-shadow'
time="     0.829" type="debug" -> gui: Loading icon theme 'oxygen'.
time="     0.829" type="debug" -> gui: Found path of base skin: ':/skins/nudus-base'.
time="     0.829" type="debug" -> gui: Base file ':/skins/nudus-base/html_wrapper.html' exists, using it for the skin.
time="     0.829" type="debug" -> gui: Local file ':/skins/nudus-light/html_style.css' exists, using it for the skin.
time="     0.830" type="debug" -> gui: Base file ':/skins/nudus-base/html_enclosure_image.html' exists, using it for the skin.
time="     0.830" type="debug" -> gui: Base file ':/skins/nudus-base/html_single_message.html' exists, using it for the skin.
time="     0.830" type="debug" -> gui: Base file ':/skins/nudus-base/html_enclosure_every.html' exists, using it for the skin.
time="     0.830" type="debug" -> gui: Local file ':/skins/nudus-light/qt_style.qss' exists, using it for the skin.
time="     0.830" type="debug" -> gui: Base file ':/skins/nudus-base/html_adblocked.html' exists, using it for the skin.
time="     0.830" type="debug" -> gui: OS defines color scheme: 'Qt::ColorScheme::Unknown'.
time="     0.830" type="debug" -> gui: Setting style: 'Fusion'.
time="     0.831" type="debug" -> gui: Skin 'nudus-light' loaded. Lite: 'false'.
time="     0.831" type="warning" -> core: User CSS style was not provided in file '/home/andrew/.config/RSS Guard 4/web/user-styles.css'.
time="     0.831" type="debug" -> network: Persistent web data storage path: '/home/andrew/.config/RSS Guard 4/web/storage'.
time="     0.831" type="debug" -> network: Disabling application-wide proxy completely.
time="     0.832" type="debug" -> core: SQLite version: '3.45.2'.
MESA-INTEL: warning: Bay Trail Vulkan support is incomplete
time="     0.847" type="debug" -> core: OpenSSL version: 'OpenSSL 3.2.1 30 Jan 2024'.
time="     0.847" type="debug" -> core: OpenSSL supported: 'true'.
time="     0.847" type="debug" -> core: Global thread pool has 4 threads.
time="     0.847" type="debug" -> core: Setting locale for LC_NUMERIC to C as libmpv requires it.
time="     0.847" type="debug" -> core: Instantiated class  'Application'.
time="     0.852" type="debug" -> database: SQLite database connection 'MessagesModel' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     0.856" type="debug" -> message-model: Repopulated model, SQL statement is now:
 'SELECT Messages.id, Messages.is_read, Messages.is_important, Messages.is_deleted, Messages.is_pdeleted, Messages.feed, Messages.title, Messages.url, Messages.author, Messages.date_created, Messages.contents, Messages.enclosures, Messages.score, Messages.account_id, Messages.custom_id, Messages.custom_hash, Feeds.title, Feeds.is_rtl, CASE WHEN LENGTH(Messages.enclosures) > 10 THEN 'true' ELSE 'false' END AS has_enclosures, (SELECT GROUP_CONCAT(Labels.name) FROM Labels WHERE Messages.labels LIKE "%." || Labels.custom_id || ".%") as msg_labels, Messages.labels FROM Messages LEFT JOIN Feeds ON Messages.feed = Feeds.custom_id AND Messages.account_id = Feeds.account_id WHERE 0 > 1;'.
time="     0.857" type="debug" -> core: Auto-download timer started with interval 10000 ms.
time="     0.857" type="debug" -> core: Creating FeedDownloader singleton.
time="     0.906" type="debug" -> gui: Creating main application form in thread: '0x740f9c51f3c0'.
time="     0.929" type="debug" -> gui: Current row changed - proxy 'QModelIndex(-1,-1,0x0,QObject(0x0))', source 'QModelIndex(-1,-1,0x0,QObject(0x0))'.
time="     0.946" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="     1.124" type="debug" -> gui: Creating tray icon menu.
time="     1.184" type="debug" -> CTRL is NOT pressed while sorting articles - sorting with standard mode.
time="     1.184" type="debug" -> CTRL is NOT pressed while sorting articles - sorting with standard mode.
time="     1.185" type="debug" -> message-model: Repopulated model, SQL statement is now:
 'SELECT Messages.id, Messages.is_read, Messages.is_important, Messages.is_deleted, Messages.is_pdeleted, Messages.feed, Messages.title, Messages.url, Messages.author, Messages.date_created, Messages.contents, Messages.enclosures, Messages.score, Messages.account_id, Messages.custom_id, Messages.custom_hash, Feeds.title, Feeds.is_rtl, CASE WHEN LENGTH(Messages.enclosures) > 10 THEN 'true' ELSE 'false' END AS has_enclosures, (SELECT GROUP_CONCAT(Labels.name) FROM Labels WHERE Messages.labels LIKE "%." || Labels.custom_id || ".%") as msg_labels, Messages.labels FROM Messages LEFT JOIN Feeds ON Messages.feed = Feeds.custom_id AND Messages.account_id = Feeds.account_id WHERE 0 > 1 ORDER BY LOWER(Messages.title) ASC, Messages.date_created DESC;'.
time="     1.193" type="debug" -> core: Hiding the main window when the application is starting.
time="     1.193" type="debug" -> database: SQLite database connection 'FeedReader' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     1.194" type="debug" -> database: SQLite database connection 'FeedlyEntryPoint' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     1.194" type="debug" -> database: SQLite database connection 'GmailEntryPoint' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     1.194" type="debug" -> database: SQLite database connection 'GreaderEntryPoint' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     1.195" type="debug" -> database: SQLite database connection 'OwnCloudServiceEntryPoint' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     1.195" type="debug" -> database: SQLite database connection 'StandardServiceEntryPoint' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     1.206" type="debug" -> database: SQLite database connection 'StandardServiceRoot' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     1.207" type="debug" -> core: Custom ID of feed when loading from DB is '4'.
time="     1.207" type="debug" -> core: Custom ID of feed when loading from DB is '8'.
time="     1.208" type="debug" -> core: Custom ID of feed when loading from DB is '9'.
time="     1.208" type="debug" -> core: Custom ID of feed when loading from DB is '10'.
time="     1.208" type="debug" -> core: Custom ID of feed when loading from DB is '11'.
time="     1.208" type="debug" -> core: Custom ID of feed when loading from DB is '12'.
time="     1.209" type="debug" -> database: SQLite database connection 'RecycleBin' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     1.210" type="debug" -> database: SQLite database connection 'ImportantNode' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     1.210" type="debug" -> database: SQLite database connection 'UnreadNode' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     1.211" type="debug" -> database: SQLite database connection 'LabelsNode' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     1.211" type="debug" -> database: SQLite connection 'StandardServiceRoot' is already active.
time="     1.211" type="debug" -> database: SQLite database connection 'StandardServiceRoot' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     1.212" type="debug" -> database: SQLite database connection 'TtRssServiceEntryPoint' to file '/home/andrew/.config/RSS Guard 4/database/database.db' seems to be established.
time="     1.212" type="debug" -> gui: User wants to have tray icon.
time="     1.212" type="warning" -> gui: Showing tray icon with little delay.
time="     1.212" type="debug" -> gui: Feed list item expanded - andrew (RSS/ATOM/JSON)
time="     1.212" type="debug" -> gui: Feed list item expanded - Labels
time="     1.212" type="debug" -> gui: Feed list item expanded - Regex queries
time="     1.213" type="debug" -> gui: Feed list item expanded - RSS Guard
time="     1.213" type="debug" -> gui: Feed list item expanded - Technology
time="     1.213" type="debug" -> gui: Feed list item expanded - Misc
time="     1.215" type="debug" -> core: No execution message received from other app instances.
time="     1.288" type="debug" -> gui: Hovered link: 'QUrl()'.
time="     1.289" type="debug" -> core: Silencing GUI message: ''.
time="     1.833" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="     2.594" type="debug" -> network: Destroying Downloader instance.
time="     2.594" type="debug" -> network: Destroying SilentNetworkAccessManager instance.
time="     4.363" type="warning" -> gui: Tray icon is available, showing now.
time="     4.370" type="debug" -> gui: Creating SystemTrayIcon instance.
time="     4.371" type="debug" -> gui: Showing tray icon immediately.
time="     4.444" type="debug" -> gui: Tray icon displayed.
time="    21.845" type="debug" -> gui: Message splitter moved.
time="    21.845" type="warning" -> gui: Some of splitter position is 0.
time="    25.650" type="warning" -> gui: Dialog 'FormSettings' was moved from 'QPoint(0,0)' to 'QPoint(248,63)'.
time="    42.349" type="debug" -> core: Syncing settings.
Segmentation fault

### Operating system and version

* OS: Arch Linux: Linux 6.8.7-artix1-2
* RSS Guard version:
Version: 4.6.6 (built on Linux/x86_64)
Revision: 
Build date: 15/03/2024 09:54
Qt: 6.7.0 (compiled against 6.6.2)

**Comment by martinrotter** [comment]
https://wiki.archlinux.org/title/Debugging/Getting_traces

**Comment by AndyM48** [comment]
Attched (I think)
[gdb.txt](https://github.com/martinrotter/rssguard/files/15147913/gdb.txt)

**Comment by martinrotter** [comment]
I see, does the crash occur everytime you do that thing or just sometimes?

**Comment by AndyM48** [comment]
Every time, no exceptions (as far as I can see)

**Comment by guihkx** [comment]
> [gdb.txt](https://github.com/martinrotter/rssguard/files/15147913/gdb.txt)

Judging by this trace:

```cpp
Thread 1 (Thread 0x7fffd3bc83c0 (LWP 25570) "rssguard"):
#0  0x00007ffff77117dd in ToastNotificationsManager::resetNotifications(bool) () at /usr/lib/librssguard.so
#1  0x00007ffff774fcc9 in SettingsNotifications::saveSettings() () at /usr/lib/librssguard.so
#2  0x00007ffff76dd500 in FormSettings::applySettings() () at /usr/lib/librssguard.so
#3  0x00007ffff76dda2e in FormSettings::saveSettings() () at /usr/lib/librssguard.so
```

It seems to be the same issue as #1317, which was fixed by commit  3b19a66dd4dbdd4408ea8e6afe9707c0d652df37.

But the fix is not present in any release yet.

@AndyM48 Can you please test with the AppImage from [development builds](https://github.com/martinrotter/rssguard/releases/tag/devbuild)? Just to be safe, make sure to back up your RSS Guard data first.

**Comment by martinrotter** [comment]
Likely fixed by the release, if not, please feel free to reopen and provide additional info.
